### ISSUE TITLE
superposition question

### ISSUE THREAD

**Issue by kenietz** [PDBeurope/pdbe-molstar#41]
Hi,
i have a question about superposition. Well maybe not superposition exactly..
How can i load 2 pdb structures which are already superposed? I use gesamt for superposition, which gives me 2 files with the coordinates calculated. In NGL i can load both files with stage.LoadFile which works fine. They are both loaded and i can select both of them. But in pdbe-molstar i could not find how i can load 2 pdb files. Parameter 'customData' accepts only one file.
I found an example for superposition at: https://embed.plnkr.co/plunk/kL2PCWp9ed1U2m99, but this one is using some internal mechanisms. It seems it is doing superpose on the fly.

I use the library as JS plugin.

I would be happy for any help or hints about where should i look at :)

Thank you for your time and efforts!

Best regards
Dimitar

**Comment by mandarsd** [comment]
Hi,
You could load additional file using the helper method - viewerInstance.visual.update()
Here is a reference -
https://embed.plnkr.co/plunk/4KpeJ6CffGiiVZlk

Though you could load and view multiple files but we have a limited support with the helper methods for such views. 
So you might need to fork and extend the code for your additional use-case.

**Comment by kenietz** [comment]
Hi,

thank you very much for your help!

I will take a look and see how to proceed :)

Best
Dimitar

**Comment by kenietz** [comment]
Hi,

now i can load two structures fine. Even managed to color them separately.
Thank you!

Howeve, I have another question. I have been trying to use the select and
during that provide representation type. This works. However the colors are
not applied to the representation itself. Say i select AA index number 133
and provide colors for the AA and the representation like this:
var data = [
{
      struct_asym_id: chain,
      start_residue_number: aaInd,
      end_residue_number: aaInd,
      representation:'ball-and-stick',
      color:colorAA,
      representationColor: colorAA
},
];
             viewerInstance.visual.select({
data: data,
nonSelectedColor: colorCH,
structureNumber: structNum
})

Select finishes without complain. However, only the color on the backbone
changes but not the color of the representation. The 'ball-and-stick' are
in default theme.

I've been diggin quite a bit and found only this link:
https://githubmemory.com/repo/PDBeurope/pdbe-molstar/issues/12, where it
said that this issue should be resolved or am i wrong?

So basically, how can i change the color of the representation as well?

Thank you for your time and efforts in advance!

Best regards
Dimitar

**Comment by mandarsd** [comment]
Hi,
There was a minor issue with the additional represetation selection which I've fixed. Please use v1.2.1, the `representationColor ` param should work properly now.

**Comment by kenietz** [comment]
Hi again,

yes there was some indexing problem it seemed. Because it was coloring the
previous representation somehow. Now in 1.2.1 works correctly but i have
another issue now. Namely 1.2.1 works if i download it from CDN but if i
git clone anew, as i did, npm install finishes ok but npm run build exits
with an error.
Here is output of the command:


> ***@***.*** build
> npm run build-tsc && npm run build-extra && npm run build-webpack && npm
run bundle-webcomponent


> ***@***.*** build-tsc
> concurrently "tsc --incremental" "tsc --build tsconfig.json --incremental"

[0] tsc --incremental exited with code 0
[1] tsc --build tsconfig.json --incremental exited with code 0

> ***@***.*** build-extra
> cpx "src/**/*.{scss,html,ico}" lib/


> ***@***.*** build-webpack
> webpack --mode production --config ./webpack.config.js

Hash: 4d51ceded18a6154f36ba23c9171783b5bf1011f
Version: webpack 4.43.0
Child
    Hash: 4d51ceded18a6154f36b
    Time: 44ms
    Built at: 12/05/2021 12:37:04 PM

    ERROR in Entry module not found: Error: Can't resolve
'/mnt/exthdd/pdbe-molstar/lib/index.js' in '/mnt/exthdd/pdbe-molstar'
Child
    Hash: a23c9171783b5bf1011f
    Time: 1692ms
    Built at: 12/05/2021 12:37:06 PM
                                        Asset      Size  Chunks
       Chunk Names
        pdbe-molstar-component-build-1.2.1.js  30.6 KiB       0  [emitted]
      main
    pdbe-molstar-component-build-1.2.1.js.map   163 KiB       0  [emitted]
[dev]  main
    Entrypoint main = pdbe-molstar-component-build-1.2.1.js
pdbe-molstar-component-build-1.2.1.js.map
    [0] ./src/web-component/index.js + 17 modules 118 KiB {0} [built]
        | ./src/web-component/index.js 8.76 KiB [built]
        |     + 17 hidden modules

**Comment by kenietz** [comment]
Hi again,

can i get some help please?

I am trying to implement a function in index.ts which clears a single
selection. That is because clearSelection clears all which i do not want to
happen in my yse case.
So far, I have the following code:

        clearSingleSelection: async (structureNumber: number, repIndex:
number) => {
var structIndex=structureNumber;
const
struct=this.plugin.managers.structure.hierarchy.current.structures[structIndex];
const comps=struct.components;
const comp=comps[repIndex];
const cell=comp.cell;
let selReprCells: any = [];
selReprCells.push(cell);
for await (const selReprCell of selReprCells) {
await clearStructureOverpaint(this.plugin, [comp]);
                        await
PluginCommands.State.RemoveObject(this.plugin, { state:
selReprCell.parent!, ref: selReprCell.transform.ref });
                };
        }

Well it does remove the requested representation but I have a problem with
function 'clearStructureOverpaint'. It does not complain in the way i
use it but also does not remove the color of the structure at the selected
AA position. If i pass the all 'structure.components' it works but it does
remove the color from all representations which is not the point of the
exercise :)

So, i am not sure how to go about it. How to change the color of the
structure at the required AA position?

Any hints and help are appreciated!!

Best wishes
Dimitar
